### PR TITLE
Add Constraint Violation Threshold as an command line option

### PR DESCRIPTION
**Pull request by anwu1219**
Give users the ability to change the constraint violation threshold.
Usage:
`./Marabou [network] [property] --split-threshold=1`
On ACAS benchmarks (property 1-4) with a 30 minutes timeout, 
split-threshold=1 solves 9 more SAT and 5 more UNSAT than split-threshold=20.

### CODE REVIEW

File: src/configuration/OptionParser.cpp [hunk 1]
```diff
@@ -76,6 +76,9 @@ void OptionParser::initialize()
           boost::program_options::value<int>( &((*_intOptions)[Options::VERBOSITY]) ),
           "Verbosity of engine::solve(). 0: does not print anything (for DnC), 1: print"
           "out statistics in the beginning and end, 2: print out statistics during solving." )
+        ( "split-threshold",
+          boost::program_options::value<int>( &((*_intOptions)[Options::SPLIT_THRESHOLD]) ),
+          "Threshold to split" )
```
Comment (ahmed-irfan) [inline src/configuration/OptionParser.cpp]: add more explanation

Comment (ahmed-irfan) [reply src/configuration/OptionParser.cpp]: perhaps "Max number of tries to fix a relu before splitting"

File: src/configuration/Options.cpp [hunk 2]
```diff
@@ -53,6 +53,7 @@ void Options::initializeDefaultValues()
     _intOptions[INITIAL_TIMEOUT] = 5;
     _intOptions[VERBOSITY] = 2;
     _intOptions[TIMEOUT] = 0;
+    _boolOptions[SPLIT_THRESHOLD] = 20;
```
Comment (ahmed-irfan) [inline src/configuration/Options.cpp]: why boolOption?

File: src/engine/Marabou.cpp [hunk 3]
```diff
@@ -102,6 +102,12 @@ void Marabou::prepareInputQuery()
             printf( "Property: None\n" );
 
         printf( "\n" );
+
+        /*
+          Step 3: extract options
+        */
+        unsigned splitThreshold = Options::get()->getInt( Options::SPLIT_THRESHOLD );
```
Comment (ahmed-irfan) [inline src/engine/Marabou.cpp]: getInt can return negative value. Need to guard that case

**Comment by guykatzz** [comment]
Other than Ahmed's comments, LGTM.

File: src/configuration/OptionParser.cpp (hunk 1 above)
Comment (anwu1219) [reply src/configuration/OptionParser.cpp]: changed comments

File: src/configuration/Options.cpp (hunk 2 above)
Comment (anwu1219) [reply src/configuration/Options.cpp]: whoops. Changed to intOptions

File: src/engine/Marabou.cpp (hunk 3 above)
Comment (anwu1219) [reply src/engine/Marabou.cpp]: guarded

**Pull request merged**